samza-core/src/main/java/org/apache/samza/execution/JobNode.java

```diff
@@ -19,361 +19,150 @@
-    serdes.forEach(serde -> {
-        String serdeName = serdeUUIDs.computeIfAbsent(serde,
-            s -> serde.getClass().getSimpleName() + "-" + UUID.randomUUID().toString());
-        configs.putIfAbsent(String.format(SerializerConfig.SERDE_SERIALIZED_INSTANCE(), serdeName),
-            base64Encoder.encodeToString(serializableSerde.toBytes(serde)));
-      });
-
-    // set key and msg serdes for streams to the serde names generated above
-    streamKeySerdes.forEach((streamId, serde) -> {
-        String streamIdPrefix = String.format(StreamConfig.STREAM_ID_PREFIX(), streamId);
-        String keySerdeConfigKey = streamIdPrefix + StreamConfig.KEY_SERDE();
-        configs.put(keySerdeConfigKey, serdeUUIDs.get(serde));
-      });
-
-    streamMsgSerdes.forEach((streamId, serde) -> {
-        String streamIdPrefix = String.format(StreamConfig.STREAM_ID_PREFIX(), streamId);
-        String valueSerdeConfigKey = streamIdPrefix + StreamConfig.MSG_SERDE();
-        configs.put(valueSerdeConfigKey, serdeUUIDs.get(serde));
-      });
-
-    // set key and msg serdes for stores to the serde names generated above
-    storeKeySerdes.forEach((storeName, serde) -> {
-        String keySerdeConfigKey = String.format(StorageConfig.KEY_SERDE(), storeName);
-        configs.put(keySerdeConfigKey, serdeUUIDs.get(serde));
-      });
-
-    storeMsgSerdes.forEach((storeName, serde) -> {
-        String msgSerdeConfigKey = String.format(StorageConfig.MSG_SERDE(), storeName);
-        configs.put(msgSerdeConfigKey, serdeUUIDs.get(serde));
+        reachableOperators.add(op);
+        findReachableOperators(op.getRegisteredOperatorSpecs(), reachableOperators);
       });
   }
 
-  /**
-   * Computes the triggering interval to use during the execution of this {@link JobNode}
-   */
-  private long computeTriggerInterval() {
-    // Obtain the operator specs from the specGraph
-    Collection<OperatorSpec> operatorSpecs = specGraph.getAllOperatorSpecs();
-
-    // Filter out window operators, and obtain a list of their triggering interval values
-    List<Long> windowTimerIntervals = operatorSpecs.stream()
-        .filter(spec -> spec.getOpCode() == OperatorSpec.OpCode.WINDOW)
-        .map(spec -> ((WindowOperatorSpec) spec).getDefaultTriggerMs())
-        .collect(Collectors.toList());
-
-    // Filter out the join operators, and obtain a list of their ttl values
-    List<Long> joinTtlIntervals = operatorSpecs.stream()
-        .filter(spec -> spec instanceof JoinOperatorSpec)
-        .map(spec -> ((JoinOperatorSpec) spec).getTtlMs())
-        .collect(Collectors.toList());
-
-    // Combine both the above lists
-    List<Long> candidateTimerIntervals = new ArrayList<>(joinTtlIntervals);
-    candidateTimerIntervals.addAll(windowTimerIntervals);
-
-    if (candidateTimerIntervals.isEmpty()) {
-      return -1;
+  // get all next operators consuming from the input {@code streamId}
+  Set<String> getNextOperatorIds(String streamId) {
```

Comment:
Minor: Prefer to order methods in class by visibility (constructors, public, protected, package private, private)

Reply:
Sure.